content/guide/agents-cloudify-agent.md

```diff
@@ -0,0 +1,188 @@
+The `init_script` and `provided` methods make use of an initialization script that gets executed on the host when it gets created; This usually takes place during the `create` operation of the cloud specific plugin used (e.g. in Openstack, `userdata` is used).
+
+To configure which installation method will be used, set the `install_method` property in the compute node `agent_config` property. For example:
+
+{{< gsHighlight  yaml  >}}
+node_templates:
+  my_vm:
+    type: cloudify.nodes.Compute
+    property:
+      agent_config:
+        # one of none, remote, init_script, provided
+        install_method: remote
+{{< /gsHighlight >}}
+
+Following are all the supported installation methods:
+
+* `none` - No agent will be installed on the host.
+* `remote` - An agent will be installed using SSH on linux hosts and WinRM on windows hosts.
+* `init_script` - An agent will be installed via a script that will run on the host when it gets created. This method is only supported for specific IaaS plugins.
+* `provided` - An agent is assumed to already be installed on the host image. That agent will be configured and started via a script that will run on the host when it gets created. This method is only supported for specific IaaS plugins.
+
+
+## Pre-requisites for Linux Remote Agent Installation
+
+* SSH port (22 by default) should be open for incoming connections.
+* An SSH server should be running on the host.
+
+## Pre-requisites for Windows Remote Agent Installation
+
+* WinRM port (5985 by default) should be open for incoming connections.
+* WinRM should be enabled. Run these command in userdata (or something equivalent) or create an image with this configuration:
+
+{{< gsHighlight  bash  >}}
+winrm quickconfig -q
+winrm set winrm/config              '@{MaxTimeoutms="1800000"}'
+winrm set winrm/config/winrs        '@{MaxMemoryPerShellMB="300";MaxShellsPerUser="2147483647"}'
+winrm set winrm/config/service      '@{AllowUnencrypted="true";MaxConcurrentOperationsPerUser="4294967295"}'
+winrm set winrm/config/service/auth '@{Basic="true"}'
+&netsh advfirewall firewall add rule name="WinRM 5985" protocol=TCP dir=in localport=5985 action=allow
+{{< /gsHighlight >}}
+
+{{% gsNote title="Note" %}}
+Note that the previous commands are very permisive and should adjusted according to your requirements.
+
+These settings provide unencrypted WinRM access to the machine. We're working on adding Kerberos support.
+From MSDN: AllowUnencrypted - Allows the client computer to request unencrypted traffic.
+{{% /gsNote %}}
+
+## Pre-requisites for Init Script and Provided Agent Installations
+
+To use the `init_script` and `provided` installation methods, an IaaS plugin that supports it should be used. At the moment, only the openstack plugin supports
+these installation methods.
+
+
+# Configuration Locations
+
+Agent configuration is composed of several locations which all adhere to the same schema. The schema is based on the `cloudify.datatypes.AgentConfig`
```

Comment:
I think it is important enough to find a way to either put the AgentConfig data type here or somehow link to it.
What do you think?